src/server/templates/typescript.ts

```diff
@@ -390,11 +392,18 @@ export type Database = {
     })}
 }
 
-type PublicSchema = Database[Extract<keyof Database, "public">]
+${
+  defaultSchema
+    ? `export type DefaultSchema = Database[Extract<keyof Database, ${JSON.stringify(defaultSchema)}>]
+  export type DefaultSchemaOrPublic = DefaultSchema`
+    : `export type PublicSchema = Database[Extract<keyof Database, "public">]
+  export type DefaultSchemaOrPublic = PublicSchema`
```

Comment:
This should be kept private and unexported

Reply:
This is now refactored to 
```ts
type DefaultSchema = Database[Extract<keyof Database, ${JSON.stringify(DEFAULT_SCHEMA) || '"public"'}>]
```